lib/charms/tls_certificates_interface/v0/tls_certificates.py

```diff
@@ -449,6 +449,40 @@ def _parse_certificates_from_relation_data(relation_data: dict) -> List[Cert]:
                     )
         return certificates
 
+    def get_certificates_for_common_name(
+        self, common_name: str, departed_id: int = None
+    ) -> List[Cert]:
+        """Loops over all relations and returns list of Cert objects.
+
+        Args:
+            common_name: return certificates for specified the common name.
+            departed_id: don't return certificates from departed relation. The argument
+                         should be set if the function is executed on relation departed
+                         event to avoid including data from the departed relation.
+
+        Returns:
+            list: List of certificates.
+        """
+        certificates = []
+        relations = self.model.relations[self.relationship_name]
+        for relation in relations:
+            if relation.id == departed_id:
```

Comment:
Please add a test for this condition, it seems important according to the doc-string and nothing is validating the functionality.